### ISSUE TITLE
Lidarr Post-Processing Failure

### ISSUE THREAD

**Issue by breezytm** [clinton-hall/nzbToMedia#1404]
I am tired of pulling my hair out. I don't see what it is that I am doing wrong. Every time I am trying to introduce a new application I have to go through this struggle. Can you please help me

`INFO	Tue Dec 11 2018 00:54:35	Collection Chris_Brown-Royalty-Deluxe_Edition-CD-FLAC-2015-PERFECT added to history
ERROR	Tue Dec 11 2018 00:54:35	Post-process-script nzbToMedia/nzbToLidarr.py for Chris_Brown-Royalty-Deluxe_Edition-CD-FLAC-2015-PERFECT failed (terminated with unknown status)
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: Exception TypeError: "'NoneType' object is not callable" in <bound method PosixProcess.__del__ of <core.nzbToMediaUtil.PosixProcess object at 0x7f52c92e5890>> ignored
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: First error at line 194.
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: configobj.configobj.ConfigObjError: Parsing failed with several errors.
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: raise error
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: File "/opt/nzbget/scripts/nzbToMedia/libs/configobj/configobj.py", line 1318, in _load
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: self._load(infile, configspec)
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: File "/opt/nzbget/scripts/nzbToMedia/libs/configobj/configobj.py", line 1229, in __init__
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: super(configobj.ConfigObj, self).__init__(*args, **kw)
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: File "/opt/nzbget/scripts/nzbToMedia/core/nzbToMediaConfig.py", line 94, in __init__
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: CFG_NEW = config()
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: File "/opt/nzbget/scripts/nzbToMedia/core/nzbToMediaConfig.py", line 253, in addnzbget
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: CFG = config.addnzbget()
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: File "/opt/nzbget/scripts/nzbToMedia/core/__init__.py", line 301, in initialize
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: core.initialize(section)
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: File "/opt/nzbget/scripts/nzbToMedia/nzbToMedia.py", line 760, in main
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: result = nzbToMedia.main(sys.argv, section)
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: File "/opt/nzbget/scripts/nzbToMedia/nzbToLidarr.py", line 244, in <module>
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: Traceback (most recent call last):
INFO	Tue Dec 11 2018 00:54:35	nzbToLidarr: [00:54:35] [ERROR]::MAIN: Unable to migrate config file /opt/nzbget/scripts/nzbToMedia/autoProcessMedia.cfg, exiting ...`

**Comment by clinton-hall** [comment]
did you manually edit the autoProcessMedia.cfg file, or was it all setup through NZBGet WebUI?

**Comment by breezytm** [comment]
Yes I did. The Lidarr section was missing so I added it.

**Comment by clinton-hall** [comment]
Well something you added was not right. Possibly bad formatting etc.

You should not add the section. This should be created when you run the script (even run it manually). You then juts edit the appropriate parameters.
Or if you run NZBGet you do it all from the web UI.

In this case, if the Lidarr section didn't exist when you ran the script, there is a bigger issue going on.

Try deleting the Lidarr section, manually run the script and see if it runs to completions. if so, it should add the Lidarr section for you to then edit the config.

**Comment by breezytm** [comment]
I backed up the file before adding the Lidarr section. I restored the back up and override the file but same problem. I haven’t had a chance to look at it today. I will again in a few

**Comment by clinton-hall** [comment]
Can you check Lidarr logs at Wed Dec 12 2018 00:23:24 to see what was logged there?

**Comment by breezytm** [comment]
`18-12-12 00:23:19.0|Debug|Nzbget|Downloaded nzb for release 'Beyonce-Lemonade-CD-FLAC-2016-FATHEAD' finished (48665 bytes from https://nzb.cat/getnzb/19a8ecd24faf0d5de324eb55180dc2231af368c5.nzb&i=8191&r=e771fd9a177e2088c5dbe7a1d9c08a2d)
18-12-12 00:23:19.0|Info|Nzbget|Adding report [Beyonce-Lemonade-CD-FLAC-2016-FATHEAD] to the queue.
18-12-12 00:23:19.2|Info|DownloadService|Report sent to NZBGet. Beyonce-Lemonade-CD-FLAC-2016-FATHEAD
18-12-12 00:23:19.2|Debug|NotificationService|No tags set for this notification.
18-12-12 00:23:19.2|Debug|Api|[POST] /api/v1/release: 200.OK (388 ms)
18-12-12 00:23:24.1|Debug|Api|[GET] /api/v1: 401.Unauthorized (0 ms)
18-12-12 00:23:24.2|Fatal|LidarrErrorPipeline|Request Failed. POST /api/v1/command

[v0.5.0.583] System.InvalidOperationException: Sequence contains no matching element
  at System.Linq.Enumerable.Single[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x00070] in <abeb153532fb46718501a600866a8d76>:0 
  at Lidarr.Api.V1.Commands.CommandModule.StartCommand (Lidarr.Api.V1.Commands.CommandResource commandResource) [0x00022] in <bc875e4e249346acbbc4c37c8fec863f>:0 
  at Lidarr.Http.REST.RestModule`1[TResource].<set_CreateResource>b__42_0 (System.Object options) [0x0000d] in <5d73e9fd9acd4ec6841b72d590c9af3e>:0 
  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func`2<object, object>,object)
  at Nancy.Routing.Route+<>c__DisplayClass4.<Wrap>b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in <bb3027f50b35411088f45475912cc2ff>:0 

18-12-12 00:23:24.2|Debug|Api|[POST] /api/v1/command: 500.InternalServerError (4 ms)
18-12-12 00:23:24.2|Debug|Parser|Parsing string 'Beyonce-Lemonade-CD-FLAC-2016-FATHEAD'
18-12-12 00:23:24.2|Debug|Parser|Album Parsed. Beyonce - Lemonade 
18-12-12 00:23:24.2|Debug|Parser|Language parsed: English
18-12-12 00:23:24.2|Debug|QualityParser|Trying to parse quality for Beyonce-Lemonade-CD-FLAC-2016-FATHEAD
18-12-12 00:23:24.2|Debug|Parser|Quality parsed: FLAC v1
18-12-12 00:23:24.2|Debug|Parser|Release Group parsed: FATHEAD
18-12-12 00:23:24.2|Debug|TrackedDownloadService|Tracking 'NZBGet:Beyonce-Lemonade-CD-FLAC-2016-FATHEAD': ClientState=Paused SonarrStage=Downloading Episode='Beyonce - Lemonade FLAC v1' OutputPath=.
18-12-12 00:23:26.0|Debug|Api|[GET] /api/v1/command: 200.OK (4 ms)
18-12-12 00:23:26.0|Debug|Api|[GET] /api/v1/health: 200.OK (3 ms)
18-12-12 00:23:26.0|Debug|Api|[GET] /api/v1/queue/status: 200.OK (5 ms)
18-12-12 00:24:09.9|Debug|Api|[GET] /api/v1/command: 200.OK (1 ms)
18-12-12 00:24:10.0|Debug|Api|[GET] /api/v1/track?albumId=15: 200.OK (5 ms)
18-12-12 00:24:10.0|Debug|Api|[GET] /api/v1/trackFile?albumId=15: 200.OK (11 ms)
18-12-12 00:24:10.0|Debug|Api|[GET] /api/v1/health: 200.OK (3 ms)
18-12-12 00:24:10.0|Debug|Api|[GET] /api/v1/queue/status: 200.OK (5 ms)
18-12-12 00:24:30.7|Debug|Parser|Parsing string 'Beyonce-Lemonade-CD-FLAC-2016-FATHEAD'
18-12-12 00:24:30.7|Debug|Parser|Album Parsed. Beyonce - Lemonade 
18-12-12 00:24:30.7|Debug|Parser|Language parsed: English
18-12-12 00:24:30.7|Debug|QualityParser|Trying to parse quality for Beyonce-Lemonade-CD-FLAC-2016-FATHEAD
18-12-12 00:24:30.7|Debug|Parser|Quality parsed: FLAC v1
18-12-12 00:24:30.7|Debug|Parser|Release Group parsed: FATHEAD
18-12-12 00:24:30.7|Debug|TrackedDownloadService|Tracking 'NZBGet:Beyonce-Lemonade-CD-FLAC-2016-FATHEAD': ClientState=Failed SonarrStage=Downloading Episode='Beyonce - Lemonade FLAC v1' OutputPath=/opt/nzbget/downloads/completed/music/Beyonce-Lemonade-CD-FLAC-2016-FATHEAD.
18-12-12 00:24:30.7|Debug|RedownloadFailedDownloadService|Failed download only contains one album, searching again
18-12-12 00:24:30.7|Info|NzbSearchService|Searching 1 indexers for [BeyoncÃ© - Lemonade (2016)]
18-12-12 00:24:30.7|Debug|Newznab|Downloading Feed https://nzb.cat/api?t=search&cat=3000,3010,3020,3030,3040&extended=1&apikey=(removed)&offset=0&limit=100&q=Beyonce%20Lemonade`

**Comment by clinton-hall** [comment]
So the issue is
```
18-12-12 00:23:24.1|Debug|Api|[GET] /api/v1: 401.Unauthorized (0 ms)
18-12-12 00:23:24.2|Fatal|LidarrErrorPipeline|Request Failed. POST /api/v1/command

[v0.5.0.583] System.InvalidOperationException: Sequence contains no matching element
at System.Linq.Enumerable.Single[TSource] (System.Collections.Generic.IEnumerable1[T] source, System.Func2[T,TResult] predicate) [0x00070] in :0
at Lidarr.Api.V1.Commands.CommandModule.StartCommand (Lidarr.Api.V1.Commands.CommandResource commandResource) [0x00022] in :0
at Lidarr.Http.REST.RestModule1[TResource].<set_CreateResource>b__42_0 (System.Object options) [0x0000d] in <5d73e9fd9acd4ec6841b72d590c9af3e>:0 at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func2<object, object>,object)
at Nancy.Routing.Route+<>c__DisplayClass4.b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in :0

18-12-12 00:23:24.2|Debug|Api|[POST] /api/v1/command: 500.InternalServerError (4 ms)
```
So it appears we are not passing everything in to Lidarr correctly? But givenm the documentation, I can't see what we have done wrong.
We pass this as 
`http://localhost:8686/api/v1/command with data: {"path": "/opt/nzbget/downloads/completed/music/Beyonce-Lemonade-CD-FLAC-2016-FATHEAD", "name": "Rename"}`

**Comment by breezytm** [comment]
I saw in another post you mentioned you tested it and it worked. I wonder what did you do different.

**Comment by breezytm** [comment]
Maybe Lidarr assigns an ID to each artists and each albums. Not quite sure how would nzbtomedia script would know what the idea. 

Instruct Lidarr to rename the list of files provided

Parameters

files (int[]) List of File IDs to rename

**Comment by clinton-hall** [comment]
Yeah, I added Lidarr processing based on Soanarr/Radarr Methodology.
They don't have a documented command that will perform the post-process/rename.

I thought, once I had implemented this based on the Radarr methodology, that someone reported this as working... this must have been a false report.

There is little I can do until support for this is added at Lidarr side. There does not appear to be any manual way to tell Lidarr to rename/process a downloaded file.

I note their wiki provides: https://github.com/lidarr/Lidarr/wiki/Custom-Post-Processing-Scripts
which states
```
If you're looking to trigger a custom script in your download client to tell Lidarr when to update, you can find more details here. 
```
however all information on that page relates to scripts triggered by Lidarr AFTER the renaming. So the information does not match the description.
